R/subset.R

```diff
@@ -33,12 +33,15 @@ subset.ts <- function(x, subset=NULL, month=NULL, quarter=NULL, season=NULL, ...
     season <- na.omit(season)
     if(length(season) == 0L)
       stop("No recognizable quarters")
-    if(min(season) < 1 | max(season) > 4)
+    if(min(season) < 1L | max(season) > 4L)
       stop("Quarters must be between 1 and 4")
   }
   else if(is.null(season))
     stop("No subset specified")
-
+  else
+    if(min(season) < 1L | max(season) > 12L)
```

Comment:
In general, season can be greater than 12. For hourly data, the seasonal period is 24 for example.